acapy_plugin_toolbox/holder/v0_1/messages/pres_request_approve.py

```diff
@@ -133,30 +136,56 @@ async def handle(self, context: RequestContext, responder: BaseResponder):
-        async with ExceptionReporter(
-            responder,
-            (
-                BaseModelError,
-                IndyHolderError,
-                LedgerError,
-                StorageError,
-                WalletNotFoundError,
-            ),
-            context.message,
-        ):
-            pres_ex_record, message = await presentation_manager.create_presentation(
-                pres_ex_record,
-                {
-                    "self_attested_attributes": self.self_attested_attributes,
-                    "requested_attributes": self.requested_attributes,
-                    "requested_predicates": self.requested_predicates,
-                },
-                comment=self.comment,
-            )
+            if not await self.authorized_to_approve(
+                session, pres_ex_record, conn_record
+            ):
+                return await report_problem(
+                    responder,
+                    "Connection lacks privilege to view message.",
+                    context.message,
+                )
+
+            presentation_manager = PresentationManager(context.profile)
+            async with ExceptionReporter(
+                responder,
+                (
+                    BaseModelError,
+                    IndyHolderError,
+                    LedgerError,
+                    StorageError,
+                    WalletNotFoundError,
+                ),
+                context.message,
+            ):
+                (
+                    pres_ex_record,
+                    message,
+                ) = await presentation_manager.create_presentation(
+                    pres_ex_record,
+                    {
+                        "self_attested_attributes": self.self_attested_attributes,
+                        "requested_attributes": self.requested_attributes,
+                        "requested_predicates": self.requested_predicates,
+                    },
+                    comment=self.comment,
+                )
+
+            await responder.send(message, connection_id=conn_record.connection_id)
 
-        await responder.send(message, connection_id=conn_record.connection_id)
+            presentation_sent = PresSent(record=pres_ex_record)
+            presentation_sent.assign_thread_from(self)
+            await responder.send_reply(presentation_sent)
 
-        presentation_sent = PresSent(record=pres_ex_record)
-        presentation_sent.assign_thread_from(self)
-        await responder.send_reply(presentation_sent)
+    async def authorized_to_approve(
+        self,
+        session: ProfileSession,
+        presentation: PresExRecord,
+        connection: ConnRecord,
+    ):
+        cred_def_id_restriction = retrieve_cred_def_id_restrictions(presentation)
+        if await determine_approved_cred_def_id_set(
+            session, connection, cred_def_id_restriction
+        ):
+            return True
+        else:
+            return False
```

Comment:
This can be simplified to:
```suggestion
        return await determine_approved_cred_def_id_set(
            session, connection, cred_def_id_restriction
        )
```